aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -257,13 +244,95 @@ SparseTensor& true_divide_out_sparse_zerodim(
   return result;
 }
 
+Tensor true_divide_sparse(const Tensor& self, const Tensor& value) {
+  auto commonDtype = at::result_type(self, value);
+
+  // Ensures floating dtype
+  if (isIntegralType(commonDtype, /*includeBool=*/ true)) {
+    commonDtype = typeMetaToScalarType(c10::get_default_dtype());
+  }
+
+  Tensor result = at::empty({0}, self.options().dtype(commonDtype));
+  return true_divide_out_sparse_zerodim(result, self, value);
+}
+
 SparseTensor& true_divide_out_sparse_scalar(
     SparseTensor& result,
     const SparseTensor& dividend,
     Scalar divisor) {
   return true_divide_out_sparse_zerodim(result, dividend, wrapped_scalar_tensor(divisor));
 }
 
+// --------------------------------------------------------------------
+// floor_divide(SparseTensor, Scalar)
+// --------------------------------------------------------------------
+
+SparseTensor& floor_divide_out_sparse_zerodim(
+  SparseTensor& result,
+  const SparseTensor& dividend,
+  const Tensor& divisor) {
+  TORCH_CHECK(divisor.dim() == 0, "Sparse floor division requires a scalar or ",
+    "zero-dim dense tensor divisor (got shape ", divisor.sizes(), " for divisor)");
+  TORCH_CHECK(!divisor.is_sparse(), "Sparse floor division requires a scalar or ",
+    "zero-dim dense tensor divisor (got a sparse divisor)");
+
+  AT_ASSERT(result.is_sparse());
+  AT_ASSERT(dividend.is_sparse());
+
+  // Case 1: result and dividend are the same tensor
+  // Performs floor division in-place
+  if (is_same_tensor(result, dividend)) {
+
+    // Ensures result tensor is coalesced
+    // Note: uncoalesced tensors, whether integral or not, cannot be floor
+    // divided correctly. The integral case is the same as sparse division
+    // (above). An integer tensor with values=[3, 3] divided by 2 would produce
+    // values=[1, 1], which sum to 2 instead of 3 (=6/2).
+    // A float tensor with values=[3., 3.] floor divided by 2 would also produce
+    // values=[1., 1.] (after truncation), which sum to 2.f instead of 3.f.
+    if (!result.is_coalesced()) {
+      coalesce_(result);
+    }
+
+    result._values().floor_divide_(divisor);
+    return result;
+  }
+
+  // Case 2: result and dividend are different tensors
+  Tensor dividend_tmp = dividend;
+
+  // Ensures dividend_tmp is coalesced (see note above)
```

Comment:
nit: it's usually nicer to give the note an explicit name -- e.g. here you are going to have basically the same message three(?) times in div and the two floor_divide cases.

Reply:
Added a named note.